extension/src/web-component-service.js

```diff
@@ -0,0 +1,52 @@
+const fetch = require('node-fetch')
+const { serializeError } = require('serialize-error')
+const configLoader = require('./config/config')
+
+const config = configLoader.load()
+
+function getOriginKeys (getOriginKeysRequestObj) {
+  return callAdyen('originKeys', getOriginKeysRequestObj)
+}
+
+function getPaymentMethods (getPaymentMethodsRequestObj) {
+  return callAdyen('paymentMethods', getPaymentMethodsRequestObj)
+}
+
+async function callAdyen (endpoint, requestObj) {
+  let response
+  try {
+    response = await fetchAsync(endpoint, requestObj)
+  } catch (err) {
+    response = serializeError(err)
+  }
+  return { request: requestObj, response }
+}
+
+async function fetchAsync (endpoint, requestObj) {
+  const request = buildRequest(requestObj)
+  const response = await fetch(`${config.adyen.apiBaseUrl}/${endpoint}`, request)
+  const responseBody = await response.json()
+  // strip away sensitive data from the adyen response.
+  delete responseBody.additionalData
```

Comment:
I wonder if we might have also sensitive data in our payload requests which we both save as payment custom fields and interface interactions?

Reply:
I am not %100 sure what kind of data is sensitive. We analyzed it before and come up only for that part is sensitive, but yes we should keep in mind we would need to strip different fields later.